src/main/java/seedu/address/ui/CommandHistory.java

```diff
@@ -0,0 +1,154 @@
+package seedu.address.ui;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.LogsCenter;
+
+/**
+ * A class to encapsulate the history of commands entered by the user.
+ */
+public class CommandHistory {
+    private static final Logger logger = LogsCenter.getLogger(CommandHistory.class);
+    private final List<String> history;
+    private int pointer;
+    private String temp;
+
+    /**
+     * Constructor for the history of commands entered.
+     *
+     * @param history
+     */
+    private CommandHistory(List<String> history) {
+        logger.info(history.toString());
+        this.history = history;
+        pointer = history.size();
+    }
+
+    public static CommandHistory init() {
+        return new CommandHistory(new ArrayList<String>());
+    }
```

Comment:
related to comment in commandbox.java, I think can just this.history = new ArrayList<String>() inside the constructor, instead of having the args